src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs

```diff
@@ -709,6 +709,15 @@ private enum ToolPackType
                 implicitPackageReferences.Add(buildPackage);
             }
 
+            // Before net8.0, ILLink analyzers shipped in a separate package.
+            // Add the analyzer package with version taken from KnownILLinkPack.
+            if (normalizedTargetFrameworkVersion < new Version(8, 0) && toolPackType is ToolPackType.ILLink)
+            {
+                var analyzerPackage = new TaskItem("Microsoft.NET.ILLink.Analyzers");
+                analyzerPackage.SetMetadata(MetadataKeys.Version, packVersion);
+                implicitPackageReferences.Add(analyzerPackage);
```

Comment:
Just curious: where do we download this from? It seems it's not on NuGet... is the SDK adding some other public feeds for this?

Reply:
I think it's downloaded from NuGet: https://www.nuget.org/packages/Microsoft.NET.ILLink.Analyzers/

For internal builds the specific version would be available on an internal feed.